simple-program-video-mode.py

```diff
@@ -0,0 +1,172 @@
+
+# Copyright (C) 2021-2023  Matthieu Houdebine (mathoudebine)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# This file is a simple Python test program using the library code to display custom content on screen (see README)
+
+import os
+import signal
+import sys
+import time
+from datetime import datetime
+
+# Import only the modules for LCD communication
+from library.lcd.lcd_comm_rev_a import LcdCommRevA, Orientation
+from library.lcd.lcd_comm_rev_b import LcdCommRevB
+from library.lcd.lcd_comm_rev_c import LcdCommRevC
+from library.lcd.lcd_comm_rev_d import LcdCommRevD
+from library.lcd.lcd_simulated import LcdSimulated
+from library.log import logger
+
+# Set your COM port e.g. COM3 for Windows, /dev/ttyACM0 for Linux, etc. or "AUTO" for auto-discovery
+# COM_PORT = "/dev/ttyACM0"
+# COM_PORT = "COM5"
+COM_PORT = "COM4"
+
+# Display revision:
+# - A      for Turing 3.5" and UsbPCMonitor 3.5"/5"
+# - B      for Xuanfang 3.5" (inc. flagship)
+# - C      for Turing 5"
+# - D      for Kipye Qiye Smart Display 3.5"
+# - SIMU   for 3.5" simulated LCD (image written in screencap.png)
+# - SIMU5  for 5" simulated LCD
+# To identify your smart screen: https://github.com/mathoudebine/turing-smart-screen-python/wiki/Hardware-revisions
+REVISION = "C"
+
+stop = False
+
+if __name__ == "__main__":
+
+    def sighandler(signum, frame):
+        global stop
+        stop = True
+
+
+    # Set the signal handlers, to send a complete frame to the LCD before exit
+    signal.signal(signal.SIGINT, sighandler)
+    signal.signal(signal.SIGTERM, sighandler)
+    is_posix = os.name == 'posix'
+    if is_posix:
+        signal.signal(signal.SIGQUIT, sighandler)
+
+    # Build your LcdComm object for Turing 5"
+    logger.info("Selected Hardware Revision C (Turing Smart Screen 5\")")
+    lcd_comm = LcdCommRevC(com_port=COM_PORT)
+
+    # Set LANDSCAPE orientation.
+    orientation = Orientation.LANDSCAPE
+    lcd_comm.SetOrientation(orientation=orientation)
+
+    # Send initialization commands
+    lcd_comm.InitializeComm()
+
+    # Set brightness in % (warning: revision A display can get hot at high brightness! Keep value at 50% max for rev. A)
+    lcd_comm.SetBrightness(level=50)
+
+    # Check if video is loaded and upload video if needed.
+    video_size = lcd_comm.GetFileSize("/mnt/SDCARD/video/particle_wave.mp4")
```

Comment:
My device doesn't have an SDCard, but there are plenty of videos on internal storage. Maybe check if there are any internal videos and randomly select one instead of trying to upload a video to the SDCARD which may not exist?

Or, split these examples out into an sdcard and an internal storage version? There may even be space available on internal storage to push the videos.

Reply:
Mine had nothing on internal storage.

You can upload a video on internal storage, with the "UploadFile" function by setting the argument "destination_path".

For example: destination_path = "/root/video/video.mp4"

I don't know if it's possible to download video from the device's internal storage. The software provided by the manufacturer doesn't have that feature.